services/streammanager/eventbridge/eventbridgemanager.go

```diff
@@ -0,0 +1,102 @@
+package eventbridge
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+// Config is the config that is required to send data to EventBridge
+type Config struct {
+	Region       string
+	AccessKeyID  string
+	AccessKey    string
+	EventBusName string
+	ResourceID   []map[string]interface{}
+	DetailType   string
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destinationConfig interface{}) (eventbridge.EventBridge, error) {
+	config := Config{}
+
+	jsonConfig, err := json.Marshal(destinationConfig)
+	err = json.Unmarshal(jsonConfig, &config)
+	if err != nil {
+		return eventbridge.EventBridge{}, fmt.Errorf("EventBridge: Error while unmarshalling destination config : %v", err.Error())
```

Comment:
Use %w. We want to standardize that for errors going forward.

https://blog.golang.org/go1.13-errors 
`Wrapping errors with %w`

Reply:
Yeah this is better...done